platform/viewer/src/lib/DICOMSR/retrieveDataFromSR.js

```diff
@@ -0,0 +1,71 @@
+import * as dcmjs from 'dcmjs';
+import { getAllDisplaySets, getInstanceMetadata } from './srUtils';
+
+const retrieveDataFromSR = Part10SRArrayBuffer => {
+  const allDisplaySets = getAllDisplaySets();
+
+  // Get the dicom data as an Object
+  const dicomData = dcmjs.data.DicomMessage.readFile(Part10SRArrayBuffer);
+  const dataset = dcmjs.data.DicomMetaDictionary.naturalizeDataset(
+    dicomData.dict
+  );
+
```

Comment:
Here is where I managed to work around the string/number mapping problem in dcmjs like this:

```
+  // This is a hack for the problem where numeric values stored as DS (Decimal String)
+  //  are being parsed as strings instead of numbers. See https://github.com/dcmjs-org/dcmjs/issues/53
+  //  for more details.
+  function fixNumericValues(o) {
+    Object.keys(o).forEach(function (k) {
+        if (o[k] !== null && typeof o[k] === 'object') {
+            fixNumericValues(o[k]);
+        }
+
+        if (k == "NumericValue" && o[k] !== null && typeof o[k] === 'string') {
+            o[k] = Number(o[k]);
+        }
+    });
+  }
+  fixNumericValues(dataset);
```